diff -ruNp a/testing/fulltests/support/simple_eval_tools.sh b/testing/fulltests/support/simple_eval_tools.sh
--- a/testing/fulltests/support/simple_eval_tools.sh	2024-02-26 14:36:03.641432345 +0100
+++ b/testing/fulltests/support/simple_eval_tools.sh	2024-02-26 14:38:15.946855878 +0100
@@ -525,7 +525,6 @@ STARTPROG() {
     if test -f $CFG_FILE; then
 	COMMAND="$COMMAND -C -c $CFG_FILE"
     fi
-    COMMAND="$COMMAND -f"
     if [ "x$PORT_SPEC" != "x" ]; then
         COMMAND="$COMMAND $PORT_SPEC"
     fi
@@ -537,10 +536,13 @@ STARTPROG() {
         OUTPUTENVVARS $LOG_FILE.command
         echo $COMMAND >> $LOG_FILE.command
     fi
-    {
-	{ $COMMAND; } >$LOG_FILE.stdout 2>&1
-	echo $? >$LOG_FILE.exitcode
-    } &
+    if [ "x$OSTYPE" = "xmsys" ]; then
+      $COMMAND > $LOG_FILE.stdout 2>&1 &
+      ## COMMAND="cmd.exe //c start //min $COMMAND"
+      ## start $COMMAND > $LOG_FILE.stdout 2>&1
+    else
+      $COMMAND > $LOG_FILE.stdout 2>&1
+    fi
 }
 
 #------------------------------------ -o-
