diff -urNp a/snmplib/cert_util.c b/snmplib/cert_util.c
--- a/snmplib/cert_util.c	2021-12-09 08:45:23.217942229 +0100
+++ b/snmplib/cert_util.c	2021-12-09 08:46:56.567562352 +0100
@@ -1368,8 +1368,7 @@ _add_certfile(const char* dirname, const
 
            okey = PEM_read_bio_PrivateKey(certbio, NULL, NULL, NULL);
            if (NULL == okey)
-               snmp_log(LOG_ERR, "error parsing key file %s\n",
-                     key->info.filename);
+               snmp_log(LOG_ERR, "error parsing key file %s\n", filename);
            else {
                key = _add_key(okey, dirname, filename, index);
                if (NULL == key) {
